mlton/ssa/ssa-tree2.fun

```diff
@@ -360,14 +368,13 @@ structure Type =
             val default = fn () =>
                (default ()) handle BadPrimApp => false
 
-            datatype z = datatype Prim.Name.t
             fun arg i = Vector.sub (args, i)
             fun oneArg f = 1 = Vector.length args andalso f (arg 0)
             fun twoArgs f = 2 = Vector.length args andalso f (arg 0, arg 1)
             fun fiveArgs f = 5 = Vector.length args andalso f (arg 0, arg 1, arg 2, arg 3, arg 4)
             val seqIndex = word (WordSize.seqIndex ())
          in
-            case Prim.name prim of
+            case primName of
```

Comment:
It seems that all of the changes in this file can be reverted with IntInf_quotRem returning a vector (rather than side-effecting refs).

Reply:
You're right; sorry, guess I missed one revert.

Reply:
Done.